### ISSUE TITLE
unnecessary extra return type in azsecret's NewClient method.

### ISSUE THREAD

**Issue by umeshbhatms** [Azure/azure-sdk-for-go#19817]
### Bug Report

- import path of package in question, 
`"github.com/Azure/azure-sdk-for-go/sdk/keyvault/azsecrets`
- SDK version
```latest```

- What happened?
This method NewClient has, I believe, unnecessary return parameter of error type.
- What did you expect or want to happen?
No extra return parameter, just the instance of the Client

**Comment by jhendrixMSFT** [comment]
As an aside, I believe that when we add support for distributed tracing, the `error` will no longer be superfluous.

**Comment by umeshbhatms** [comment]
Thanks for the input! @chlowell , @jhendrixMSFT  I raised this issue because while conducting unit tests for my Go project, I was unable to achieve 100% code coverage due to an untriggered condition that could result in an error for the NewClient method. One solution could have been to use a "_" variable and ignore the error, but as you mentioned, it could potentially have a proper error value in the future. Therefore, I kept the error variable as is in my code, but this makes the error scenario not completely testable.
